src/engine/Operation.cpp

```diff
@@ -501,6 +501,18 @@ void Operation::updateRuntimeInformationOnFailure(Milliseconds duration) {
   signalQueryUpdate();
 }
 
+// __________________________________________________________________
+void Operation::applyLimit(const LimitOffsetClause& limitOffsetClause) {
+  _limit._offset += limitOffsetClause._offset;
+  if (limitOffsetClause._limit.has_value()) {
+    _limit._limit =
+        std::min(_limit.limitOrDefault(), limitOffsetClause._limit.value());
+  }
```

Comment:
This can be a member function of the `Limit|OffsetClause` like `add LImit`
in particular: If both are nullopt, then stay at nullopt, this gives the most consistent behavior.